providers/scaleway/v1/flags.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	DefaultPool = "default"
+
+	category   = "Scaleway"
+	flagPrefix = "scw"
+	envPrefix  = "WOODPECKER_SCW"
+)
+
+var ProviderFlags = []cli.Flag{
+	&cli.StringFlag{
+		Name:    flagPrefix + "-access-key",
+		Usage:   "Scaleway IAM API Token Access Key",
+		EnvVars: []string{envPrefix + "_ACCESS_KEY"},
+		// NB(raskyld): We should recommend the usage of file-system to users
+		// Most container runtimes support mounting secrets into the fs
+		// natively.
+		FilePath: os.Getenv(envPrefix + "_ACCESS_KEY_FILE"),
+		Category: category,
+	},
+	&cli.StringFlag{
+		Name:    flagPrefix + "-secret-key",
+		Usage:   "Scaleway IAM API Token Secret Key",
+		EnvVars: []string{envPrefix + "_SECRET_KEY"},
+		// NB(raskyld): We should recommend the usage of file-system to users
+		// Most container runtimes support mounting secrets into the fs
+		// natively.
+		FilePath: os.Getenv(envPrefix + "_SECRET_KEY_FILE"),
+		Category: category,
+	},
+	// TODO(raskyld): implement multi-AZ
+	&cli.StringFlag{
+		Name:     flagPrefix + "-zone",
+		Usage:    "Scaleway Zone where to spawn instances",
+		EnvVars:  []string{envPrefix + "_ZONE"},
+		Category: category,
+		Value:    scw.ZoneFrPar2.String(),
+	},
+	&cli.StringFlag{
+		Name:     flagPrefix + "-instance-type",
+		Usage:    "Scaleway Instance type to spawn",
+		EnvVars:  []string{envPrefix + "_INSTANCE_TYPE"},
+		Category: category,
+	},
+	&cli.StringSliceFlag{
+		Name:     flagPrefix + "-tags",
+		Usage:    "Comma separated list of tags to uniquely identify the instances spawned",
+		EnvVars:  []string{envPrefix + "_TAGS"},
+		Category: category,
+	},
+	&cli.StringFlag{
+		Name:     flagPrefix + "-project",
+		Usage:    "Scaleway Project ID in which to spawn the instances",
+		EnvVars:  []string{envPrefix + "_PROJECT"},
+		Category: category,
+	},
+	&cli.StringFlag{
+		Name:     flagPrefix + "-prefix",
+		Usage:    "Prefix prepended before any Scaleway resource name",
+		EnvVars:  []string{envPrefix + "_PREFIX"},
+		Category: category,
+		Value:    "wip-woodpecker-ci-autoscaler",
+	},
+	&cli.BoolFlag{
+		Name:     flagPrefix + "-enable-ipv6",
+		Usage:    "Enable IPv6 for the instances",
+		EnvVars:  []string{envPrefix + "_ENABLE_IPV6"},
+		Category: category,
+	},
+	&cli.StringFlag{
+		Name:     flagPrefix + "-image",
+		Usage:    "The base image for your instance",
+		EnvVars:  []string{envPrefix + "_IMAGE"},
+		Category: category,
+		Value:    "ubuntu_jammy",
+	},
+	&cli.Uint64Flag{
+		Name:     flagPrefix + "-storage-size",
+		Usage:    "How much storage to provision for your agents in bytes",
+		EnvVars:  []string{envPrefix + "_STORAGE_SIZE"},
+		Category: category,
+		Value:    25000000000,
+	},
+	&cli.IntFlag{
+		Name:     flagPrefix + "-client-max-retries",
+		Usage:    "How much times should we retry requests (< 0: infinite, 0: no retry)",
+		EnvVars:  []string{envPrefix + "_CLIENT_MAX_RETRIES"},
+		Category: category,
+		Value:    5,
+	},
```

Comment:
Is the whole backoff logic really needed. Looks a bit overengineered to me for the moment. If testing shows this is really a thing I would rather add it to the general autoscaler logic.

Reply:
I am fine removing the back-off logic from the provider, but network failures happens every time so in the future we should at least consider adding it to the general autoscaler logic like you suggest

Reply:
> I am fine removing the back-off logic from the provider, but network failures happens every time so in the future we should at least consider adding it to the general autoscaler logic like you suggest

Sure that could happen, but I would suggest to follow a minimalist development approach. There is always a lot that could potentially happen so lets fix those things as soon as they arise.